.github/workflows/sanity_checks_for_prs.yml

```diff
@@ -0,0 +1,170 @@
+name: sanity checks
+# merge_group triggers happen in merge queues
+# workflow_dispatch triggers happen when a user manually runs the workflow
+# pull_request triggers happen when a pull request is updated
+on: [merge_group, pull_request, workflow_dispatch]
+jobs:
+  pre-commit:
+    name: Carry out pre-commit checks
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+      - name: Install pre-commit
+        run: python -m pip install pre-commit
+      - name: Cache pre-commit environments
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit-3|${{ hashFiles('.pre-commit-config.yaml') }}
+          restore-keys: |
+            pre-commit-3|
+      - name: Run pre-commit (on all files)
+        run: pre-commit run --show-diff-on-failure --color=always --all-files
+        shell: bash
+        id: Run_pre-commit
+        # this makes sure we do not accidentally use hooks that need docker on the dev machines
+        env:
+          DOCKER_HOST: docker_intentionally_disabled
+  commit_lint:
+    name: Ensure Git etiquette
+    runs-on: ubuntu-22.04
+    timeout-minutes: 5
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+      - name: Install dependencies
+        run: pip install pygithub==1.59.0
+      - name: Block Autosquash Commits
+        if: ${{ !github.event.pull_request.draft }}
+        shell: python {0}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_REPOSITORY: ${{ github.repository }}
+          HEAD_SHA: ${{  github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.event.merge_group.head_sha }}
+          BASE_SHA: ${{  github.event_name == 'pull_request' && github.event.pull_request.base.sha || github.event.merge_group.base_sha }}
+        run: |
+          from github import Github
+          import os
+          import sys
+          repo_name = os.environ['GITHUB_REPOSITORY']
+          head_sha = os.environ['HEAD_SHA']
+          base_sha = os.environ['BASE_SHA']
+          g = Github(os.environ["GITHUB_TOKEN"])
+          blocked_words = ["fixup!", "squash!",]
+          blocked_messages = ["f", "REVERT ME BEFORE MERGE"]
+          repo = g.get_repo(repo_name)
+          head_commit = repo.get_commit(head_sha)
+          base_commit = repo.get_commit(base_sha)
+          commits = repo.compare(base_commit.sha, head_commit.sha).commits
+          for commit in commits:
+              commit_msg = commit.commit.message
```

Comment:
Is the `commit.commit.message` correct here? Looks like a commit too much on first sight.

Reply:
Sorry, I forgot to note that this a copy of https://github.com/arup-group/oasys-core-cmake/blob/main/.github/workflows/sanity_checks_for_prs.yml

Reply:
Okay, so it is probably correct?

Reply:
Yeah. It's worked for good while for us I think :)